cache/slice.go

```diff
@@ -0,0 +1,35 @@
+package cache
+
+import "io"
+
+type SliceReadWriter struct {
+	slice    [][]byte
+	position uint64
+}
+
+func (s *SliceReadWriter) Width() uint64 {
+	return uint64(len(s.slice))
+}
+
+func (s *SliceReadWriter) Seek(index uint64) error {
+	if index >= uint64(len(s.slice)) {
+		return io.EOF
+	}
+	s.position = index
+	return nil
+}
+
+func (s *SliceReadWriter) ReadNext() ([]byte, error) {
+	if s.position >= uint64(len(s.slice)) {
+		return nil, io.EOF
+	}
+	value := make([]byte, NodeSize)
+	copy(value, s.slice[s.position])
+	s.position++
+	return value, nil
+}
+
+func (s *SliceReadWriter) Append(p []byte) (n int, err error) {
```

Comment:
no need for error here

Reply:
not for a slice, but this interface is intended to be used with files, as well, where errors can occur.

Reply:
Specifically, I use it with a file in PoST.